### PR TITLE
Add support for custom preferences page

### PR DESCRIPTION
**Pull request by nt1m**

### CODE REVIEW

**Comment by kumar303** [comment]
Thanks for the patch. This looks pretty straightforward. It also needs a unit test. You can run the suite with `npm test unit`; take a look at https://github.com/mozilla-jetpack/jpm/blob/master/test/unit/test.rdf.js

**Comment by nt1m** [comment]
@kumar303 Thanks for taking a look! Added automated tests.

File: test/unit/test.rdf.js [hunk 1]
```diff
@@ -548,6 +548,96 @@ describe("lib/rdf", function() {
...
+      // Check if optionsURL is properly set
+      var optURL = xml.getElementsByTagName("em:optionsURL")[0];
+      expect(optURL.firstChild.data).to.be.equal("resource://preferences-at-jetpack/data/preferences.html");
+
+      // Check if optionsType is properly set
+      var optType = xml.getElementsByTagName("em:optionsType")[0];
+      expect(optType.firstChild.data).to.be.equal("3");
+    });
+
+    it("sets em:optionsType to 3 both `preferences` and `preferencesURL` exist", function() {
```
Comment (kumar303) [inline test/unit/test.rdf.js]: nit: typo: missing "if" in the sentence

**Comment by kumar303** [comment]
The tests look great, thanks. Can you squash it into one commit?

**Comment by nt1m** [comment]
@kumar303 Done, thanks!

**Comment by kumar303** [comment]
Hmm. I'm not sure why the suite is failing now. Do you see these same failures locally?

**Comment by nt1m** [comment]
@kumar303 I can't reproduce locally with `npm test`. It don't see how squashing commits can cause this. Can you try retriggering the Travis build ?

**Comment by kumar303** [comment]
I retriggered the build once already, didn't seem to make a difference. Did you rebase on top of the latest master? Might be worth a try.

**Comment by freaktechnik** [comment]
The failures here seem to be the same as in #546

**Comment by kumar303** [comment]
You can rebase on master to get a fix for the failing Firefox tests

**Comment by kumar303** [comment]
The commit message should also be prefixed with `feat:` (for 'feature') because we've switched to a [semantic changelog](https://github.com/mozilla-jetpack/jpm/blob/master/CONTRIBUTING.md#writing-commit-messages).

**Comment by kumar303** [comment]
moved to https://github.com/mozilla-jetpack/jpm/pull/549

**Pull request closed without merging**